Code/Framework/AzCore/AzCore/Name/NameDictionary.cpp

```diff
@@ -405,4 +374,32 @@ namespace AZ
         const uint32_t hash = AZStd::hash<AZStd::string_view>()(name) & 0xFFFFFFFF;
         return hash;
     }
+
+
+    // NameDictionary::ScopedNameDataWrapper RAII implementation
+    NameDictionary::ScopedNameDataWrapper::ScopedNameDataWrapper(NameDictionary& nameDictionary, Internal::NameData* nameData)
+        : m_nameData(nameData)
+        , m_nameDictionary(nameDictionary)
+    {
+        if (m_nameData->m_nameDictionary == nullptr)
```

Comment:
Hmm, is it safe to allow name data to be registered to multiple dictionaries? It looks like the reference counting behavior (including destroying `NameData` on dictionary destruction) is preserved. Name literals make sharing names between dictionaries a valid use-case, so I'd suggest adding a unit tests that share a name between two dictionaries in various scenarios to sanity check.

Reply:
Yeah. what should occur is that when the dictionary is destroyed it resets any registered NameData pointer to it back to nullptr.